# NovaTel Customer Retention & Revenue Intelligence System

## KPI Framework

### Objective

Translate business findings from exploratory data analysis into measurable Key Performance Indicators (KPIs) that management can use to monitor customer retention, revenue performance, churn risk, loyalty, and marketing effectiveness.

### KPI Categories

- Customer KPIs
- Revenue KPIs
- Retention KPIs
- Loyalty KPIs
- Product KPIs
- Marketing KPIs

In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

In [2]:
df = pd.read_csv("../data/raw/telecom_customer_churn.csv")

df.shape

(7043, 38)

## KPI 1: Total Customers

### Business Purpose

Represents the total customer base managed by NovaTel.

### Formula

Total Customers = Count of Unique Customer IDs

In [3]:
df["Customer ID"].nunique()

7043

### Result

Total Customers = 7,043

### Business Insight

- NovaTel serves 7,043 customers.
- This KPI represents the total active customer population available for retention and revenue generation.
- All customer-related KPIs should be evaluated relative to this customer base.

## KPI 2: Churned Customers

### Business Purpose

Measures the number of customers who have left NovaTel.

### Formula

Churned Customers = Count of Customers where Customer Status = "Churned"

In [4]:
df[df["Customer Status"] == "Churned"]["Customer ID"].nunique()

1869

### Result

Churned Customers = 1,869

### Business Insight

- NovaTel has lost 1,869 customers.
- Customer churn represents a significant risk to revenue growth and customer retention.
- Reducing churn should remain a primary business objective.

## KPI 3: Churn Rate

### Business Purpose

Measures the percentage of customers who have left NovaTel.

### Formula

Churn Rate =
(Churned Customers / Total Customers) × 100

In [5]:
round(
    (
        df[df["Customer Status"] == "Churned"]["Customer ID"].nunique()
        /
        df["Customer ID"].nunique()
    ) * 100,
    2
)

26.54

### Result

Churn Rate = 26.54%

### Business Insight

- More than one-quarter of NovaTel customers have churned.
- Customer retention represents a major opportunity for business improvement.
- Reducing churn could significantly improve revenue protection and customer lifetime value.

## KPI 4: Retention Rate

### Business Purpose

Measures the percentage of customers successfully retained by NovaTel.

### Formula

Retention Rate =
(Stayed Customers / Total Customers) × 100

In [6]:
round(
    (
        df[df["Customer Status"] == "Stayed"]["Customer ID"].nunique()
        /
        df["Customer ID"].nunique()
    ) * 100,
    2
)

67.02

### Result

Retention Rate = 67.02%

### Business Insight

- NovaTel retains 67.02% of its customers.
- Retained customers generate the majority of company revenue.
- Improving retention rates would directly support revenue growth and customer lifetime value.

## KPI 5: Revenue at Risk

### Business Purpose

Measures the total revenue associated with churned customers.

### Formula

Revenue at Risk =
Sum of Total Revenue for Churned Customers

In [7]:
round(
    df[df["Customer Status"] == "Churned"]["Total Revenue"].sum(),
    2
)

np.float64(3684459.82)

### Result

Revenue at Risk = 3,684,459.82

### Business Insight

- Churned customers account for 3.68 million in revenue.
- Customer churn is not only a retention problem but also a significant revenue problem.
- Reducing churn could protect millions in customer revenue.
- Revenue protection should be a key objective of NovaTel's retention strategy.

## KPI 6: Average Revenue Per Customer (ARPC)

### Business Purpose

Measures the average revenue generated by each customer.

### Formula

ARPC =
Total Revenue / Total Customers

In [8]:
round(
    df["Total Revenue"].sum()
    /
    df["Customer ID"].nunique(),
    2
)

np.float64(3034.38)

### Result

Average Revenue Per Customer (ARPC) = 3,034.38

### Business Insight

- The average NovaTel customer generates 3,034.38 in revenue.
- This KPI provides a benchmark for customer value.
- Customer retention initiatives should focus on protecting and increasing customer lifetime value.
- Losing a customer means losing future revenue potential beyond the current revenue already generated.

## KPI 7: Average Customer Tenure

### Business Purpose

Measures the average length of customer relationships with NovaTel.

### Formula

Average Customer Tenure =
Average of Tenure in Months

In [9]:
round(
    df["Tenure in Months"].mean(),
    2
)

np.float64(32.39)

### Result

Average Customer Tenure = 32.39 Months

### Business Insight

- The average NovaTel customer remains with the company for 32.39 months.
- Customer tenure is strongly associated with retention and loyalty.
- Increasing early-stage customer retention could improve overall customer lifetime value.
- Long-tenure customers represent NovaTel's most valuable customer segment.

## KPI 8: Average Referrals Per Customer

### Business Purpose

Measures customer advocacy and word-of-mouth engagement.

### Formula

Average Referrals =
Average of Number of Referrals

In [10]:
round(
    df["Number of Referrals"].mean(),
    2
)

np.float64(1.95)

### Result

Average Referrals Per Customer = 1.95

### Business Insight

- NovaTel customers generate an average of 1.95 referrals.
- Referral behavior is strongly associated with customer loyalty.
- Customers who refer others are significantly more likely to remain with NovaTel.
- Referral programs may support both customer acquisition and retention objectives.

## KPI 9: Competitor Churn Percentage

### Business Purpose

Measures the percentage of churn caused by competitor-related reasons.

### Formula

Competitor Churn % =
(Competitor Churn Customers / Total Churned Customers) × 100

In [11]:
round(
    (
        df[df["Churn Category"] == "Competitor"].shape[0]
        /
        df[df["Customer Status"] == "Churned"].shape[0]
    ) * 100,
    2
)

45.0

### Result

Competitor Churn Percentage = 45.00%

### Business Insight

- Competitor-related reasons account for 45% of all customer churn.
- Competitive pressure is the largest driver of customer loss.
- Device offerings, pricing, download speeds, and promotional offers appear to be key competitive weaknesses.
- Improving competitive positioning could significantly reduce overall churn.

# KPI Framework Summary

## Customer KPIs

| KPI | Value |
|------|------:|
| Total Customers | 7,043 |
| Churned Customers | 1,869 |
| Churn Rate | 26.54% |
| Retention Rate | 67.02% |

### Key Insight

- More than one in four customers have churned.
- Customer retention remains NovaTel's primary business challenge.

---

## Revenue KPIs

| KPI | Value |
|------|------:|
| Total Revenue | 21,371,131.69 |
| Revenue At Risk | 3,684,459.82 |
| Average Revenue Per Customer | 3,034.38 |

### Key Insight

- Customer churn has placed 3.68 million revenue at risk.
- Retention directly impacts financial performance.

---

## Loyalty KPIs

| KPI | Value |
|------|------:|
| Average Customer Tenure | 32.39 Months |
| Average Referrals | 1.95 |

### Key Insight

- Long-tenure customers are significantly more loyal.
- Referral activity is strongly associated with customer retention.

---

## Churn Driver KPIs

| KPI | Value |
|------|------:|
| Competitor Churn Percentage | 45.00% |
| Highest Churn Contract | Month-to-Month |
| Month-to-Month Churn Rate | 45.84% |
| Highest Churn Internet Type | Fiber Optic |
| Fiber Optic Churn Rate | 40.72% |

### Key Insight

- Competitor pressure is the largest churn driver.
- Month-to-Month customers are the highest-risk customer segment.
- Fiber Optic customers experience elevated churn risk.

---

## Marketing KPIs

| KPI | Value |
|------|------:|
| Best Offer | Offer A |
| Offer A Churn Rate | 6.73% |
| Worst Offer | Offer E |
| Offer E Churn Rate | 52.92% |

### Key Insight

- Offer performance varies significantly across customer groups.
- Offer A is highly effective for retention.
- Offer E requires immediate review and optimization.

---

# Recommended Dashboard KPI Cards

## Executive Overview

- Total Customers
- Churned Customers
- Churn Rate
- Retention Rate
- Total Revenue
- Revenue At Risk

## Retention Intelligence

- Average Customer Tenure
- Average Referrals
- Competitor Churn Percentage
- Highest Churn Contract
- Highest Churn Internet Type
- Best Offer

## Customer & Loyalty Intelligence

- Worst Offer
- Senior Customer Churn Rate
- Top Churn City
- Fiber Optic Churn Rate
- Month-to-Month Churn Rate
- Average Revenue Per Customer

---

## Strategic Conclusion

NovaTel's churn challenge is primarily driven by competitor pressure, Month-to-Month contracts, Fiber Optic customers, and low-tenure customer segments. The strongest retention opportunities include migrating customers to longer-term contracts, improving competitive positioning, optimizing offer strategy, and strengthening customer loyalty programs through referrals and targeted retention initiatives.